pkg/control/control.go

```diff
@@ -245,23 +244,32 @@ func (c *Controller) RunSelfScheduled() {
 	c.setUpClient()
 	c.checkPanicLogs()
 
-	nctx, ncancel := context.WithTimeout(c.ctx, c.cfg.RunTime*time.Duration(int64(c.cfg.RunRound)))
-	var nemesisWg sync.WaitGroup
+	var (
+		nemesisWg     sync.WaitGroup
+		g             errgroup.Group
+		nCtx, nCancel = context.WithTimeout(c.ctx, c.cfg.RunTime*time.Duration(int64(c.cfg.RunRound)))
+	)
 	nemesisWg.Add(1)
 	go func() {
 		defer nemesisWg.Done()
-		c.dispatchNemesis(nctx)
+		c.dispatchNemesis(nCtx)
 	}()
 
-	// ctx, _ := context.WithTimeout(c.ctx, c.cfg.RunTime)
-	// No matter how many clients are created, we only use one here
-	// the real multi clients logic should handle by case itself
-	err := c.clients[0].Start(nctx, c.cfg.ClientConfig, c.cfg.ClientNodes)
-	if err != nil {
-		log.Panicf("start case error, %+v", errors.ErrorStack(err))
+	for i := 0; i < len(c.clients); i++ {
+		client := c.clients[i]
+		g.Go(func() error {
+			log.Infof("run client %d...", i)
+			return client.Start(nCtx, c.cfg.ClientConfig, c.cfg.ClientNodes)
+		})
+	}
+	if err := g.Wait(); err != nil {
+		log.Fatalf("run client error, %+v", errors.ErrorStack(err))
 	}
 
-	ncancel()
+	// cancel nCtx and wait nemesis ended
+	nCancel()
+	nemesisWg.Wait()
```

Comment:
Seems this wait group is equal to a channel

Reply:
I found that we lost this wait statement in preivous.

Reply:
I just feel a little odd about wait group has only one task (it's equal to waiting for a channel), but it's ok to me.